raft/src/test/java/org/apache/kafka/raft/LeaderStateTest.java

```diff
@@ -36,30 +36,31 @@
 import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
-public class LeaderStateTest {
+public class LeaderStateTest<T> {
     private final int localId = 0;
     private final int epoch = 5;
     private final LogContext logContext = new LogContext();
 
-    private LeaderState newLeaderState(
+    private LeaderState<T> newLeaderState(
         Set<Integer> voters,
         long epochStartOffset
     ) {
-        return new LeaderState(
+        return new LeaderState<>(
             localId,
             epoch,
             epochStartOffset,
             voters,
             voters,
-            logContext
+            logContext,
+            null
```

Comment:
I would not pass a `null` and add a test checking that this field is returned correctly.

Reply:
Yeah, let's add an explicit `requireNonNull` in the constructor. Here we could potentially use a mock.